packages/menu/src/MenuItem.ts

```diff
@@ -92,26 +92,41 @@ export class MenuItem extends ActionButton {
 
     protected updated(changes: PropertyValues): void {
         super.updated(changes);
-        if (this.getAttribute('role') === 'option' && changes.has('selected')) {
-            this.setAttribute(
-                'aria-selected',
-                this.selected ? 'true' : 'false'
-            );
+        const role = this.getAttribute('role');
+        if (changes.has('selected')) {
+            if (role === 'option') {
+                this.setAttribute(
+                    'aria-selected',
+                    this.selected ? 'true' : 'false'
+                );
+            } else if (
+                role === 'menuitemcheckbox' ||
+                role === 'menuitemradio'
+            ) {
+                this.setAttribute(
+                    'aria-checked',
+                    this.selected ? 'true' : 'false'
+                );
+            }
         }
     }
 
+    public updateRole(): void {
```

Comment:
This is part of what I haven't gotten quite working yet. The idea is that the parent menu asks the item to update when it sees it's children change.

Reply:
If the parent has the data https://github.com/adobe/spectrum-web-components/pull/1475/files#diff-b9cace30081552e1a2c474c35ae00ff7a5c7abe3d7c6b4722179e06d56d4ceb0R124 and tells the child to update the data (this method), maybe we shouldn't use an event anymore?

Reply:
This is working (at least for now) with the event approach so marking this resolved.